### PR TITLE
adds gardener images to component descriptor

### PR DESCRIPTION
**Pull request by jschicktanz**
**How to categorize this PR?**

/area delivery
/kind TODO

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by jschicktanz** [comment]
@timebertt @ialidzhikov @AndreasBurger

File: .ci/resources.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (timebertt) [inline .ci/resources.yaml:1]: What about the control plane components (gardener-apiserver, etc.)?

Comment (jschicktanz) [reply .ci/resources.yaml:1]: I added the remaining images I found in a previous gardener component-descriptor, please double check again that the list is correct

**Review by timebertt — APPROVED** [review]
Thanks!
/lgtm

File: .ci/resources.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+name: 'admission-controller'
+version: '${COMPONENT_VERSION}'
+type: 'ociImage'
+relation: 'local'
+access:
+  type: 'ociRegistry'
+  imageReference: 'eu.gcr.io/gardener-project/gardener/admission-controller:${COMPONENT_VERSION}'
+---
+---
```
Comment (ialidzhikov) [inline .ci/resources.yaml:10]: Why we need this separator 2 times?

```suggestion
---
```

File: .ci/resources.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+name: 'seed-admission-controller'
+version: '${COMPONENT_VERSION}'
+type: 'ociImage'
+relation: 'local'
+access:
+  type: 'ociRegistry'
+  imageReference: 'eu.gcr.io/gardener-project/gardener/seed-admission-controller:${COMPONENT_VERSION}'
+---
```
Comment (ialidzhikov) [inline .ci/resources.yaml:63]: I guess there is no need for such separator at the end of the file.

```suggestion
```

File: hack/.ci/component_descriptor [hunk 4]
```diff
@@ -6,6 +6,34 @@ repo_root_dir="$1"
...
+
+  # component-cli expects a directory where the component descriptor file is named component-descriptor.yaml.
+  # however the pre-rendered component descriptors of the pipeline have different filenames.
+  # therefore create a tempdir and copy the pre-rendered component descriptor to it with the correct filename.
+  tmp_dir="$(mktemp -d)"
+  tmp_cd="${tmp_dir}/component-descriptor.yaml"
+  cp "${BASE_DEFINITION_PATH}" "${tmp_cd}"
+  echo "${tmp_cd}"
+
+  # read component component version.
```
Comment (ialidzhikov) [inline hack/.ci/component_descriptor:21]:
```suggestion
  # read the component version.
```

**Comment by timebertt** [comment]
From https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-master/jobs/master-head-update-job/builds/347
```
selected worker: concourse-worker-2
2022-04-19 15:27:57,554 [INFO] __main__: default component descriptor (v2):

component:
  componentReferences: []
  labels: []
  name: github.com/gardener/gardener
  provider: internal
  repositoryContexts:
  - baseUrl: eu.gcr.io/sap-se-gcr-k8s-private/cnudie/gardener/development
    componentNameMapping: urlPath
    type: ociRegistry
  resources: []
  sources:
  - access:
      commit: 703a7995522f515e4140cca63aab045447697e5a
      ref: dev-703a7995522f515e4140cca63aab045447697e5a
      repoUrl: github.com/gardener/gardener
      type: github
    extraIdentity: {}
    labels:
    - name: cloud.gardener/cicd/source
      value:
        repository-classification: main
    name: github_com_gardener_gardener
    type: git
    version: v1.45.0-dev-703a7995522f515e4140cca63aab045447697e5a
  version: v1.45.0-dev-703a7995522f515e4140cca63aab045447697e5a
meta:
  schemaVersion: v2
signatures: []




Adding additional resources from /tmp/build/60e5e9b3/git-gardener.gardener-master.master/.ci/../.ci/resources.yaml
/tmp/tmp.5avlR2UwFm/component-descriptor.yaml
cat: managed-version/version: No such file or directory
Traceback (most recent call last):
  File "<string>", line 413, in <module>
  File "/usr/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/tmp/build/60e5e9b3/git-gardener.gardener-master.master/.ci/component_descriptor']' returned non-zero exit status 1.

```

**Pull request merged**